documentation/modules/exploit/multi/http/agent_tesla_panel_rce.md

```diff
@@ -0,0 +1,268 @@
+
+The [WebPanel3.7z](https://github.com/mekhalleh/agent_tesla_panel_rce/raw/master/resources/WebPanel3.7z) should be used preferably because it is ***not protected*** by `ioncube` and because of this the source code is in clear (but all the other panels are functional with a little more complicated configuration).
+
+I am based on the file creation dates (and diff) to get an idea of ​​the panels evolution, You have more details (in french) on my blog post [Agent Tesla Remote Command Execution (fighting the WebPanel)](https://www.pirates.re/agent-tesla-remote-command-execution-fighting-the-webpanel/).
+
+***IMPORTANT:*** This version of the panel is "by default" patched against unauthenticated RCE. But as I explain in my blog, the patch simply consist adding authentication on the vulnerable page. And that the variables are still not properly sanitized.
+
+To put it simply, with this version of the panel, by default you have an 'Authenticated RCE'.
+
+Now, if you want to "switch" to test simply the `Unauthenticated RCE` with this panel, ***just put the authentication in comment*** on the `WebPanel/server_side/scripts/server_processing.php` page.
+
+As example:
+
+```php
+/* // Comment the authentication.
+session_start();
+  if (!isset($_SESSION['logged_in'])
+    || $_SESSION['logged_in'] !== true) {
+    header('Location: login.php');
+    exit;
+  }
+*/
+```
+
+1. Uncompress the [WebPanel3.7z](https://github.com/mekhalleh/agent_tesla_panel_rce/raw/master/resources/WebPanel3.7z) into your Web directory.
+2. Remove the config file `WebPanel/config.php` (it is generated at step 3).
+3. Go to: <http://localhost/WebPanel/setup.php>.
+4. You can log into Agent Tesla Web panel, all is done.
+
+## Verification Steps
+
+1. Start msfconsole
+2. Do: `use exploit/multi/http/agent_tesla_panel_rce`
+3. Do: `set RHOSTS IP`
+4. Do: `run`
+--or--
+1. Start msfconsole
+2. Do: `use exploit/multi/http/agent_tesla_panel_rce`
+3. Do: `set RHOSTS IP`
+4. Do: `set USERNAME REDACTED`
+5. Do: `set PASSWORD REDACTED`
+5. Do: `run`
+
+## Targets
+
+```
+   Id  Name
+   --  ----
+   0   Automatic (Dropper)
+   1   Unix (In-Memory)
+   2   Windows (In-Memory)
+```
+
+## Options
+
+**PASSWORD**
+
+The Agent Tesla CnC password to authenticate with (needed if you attempt an authenticated RCE).
+
+**Proxies**
+
+A proxy chain of format type:host:port[,type:host:port][...]. It's optional.
+
+**RHOSTS**
+
+The target IP address on which the control center responds.
+
+**RPORT**
+
+The target TCP port on which the control center responds. Default: 80
+
+**SSL**
+
+Negotiate SSL/TLS for outgoing connections. Default: false
+
+**TARGETURI**
+
+The base URI path of control center. Default: '/WebPanel'
+
+**USERNAME**
+
+The Agent Tesla CnC username to authenticate with (needed if you attempt an authenticated RCE).
+
+**VHOST**
+
+The target HTTP server virtual host.
+
+## Usage
+
+### Targeting Windows
```

Comment:
This is not descriptive. See https://github.com/rapid7/metasploit-framework/pull/13401/files and note how he uses the `#Scenario` section and then lists each of the targets he ran it against, including the software version, version of the host, etc. You need to include this level of detail here. `## Targeting Windows` does not tell me anything about what version you targeted. 

Also given you are only targeting three versions of this software here, I would expect to see three scenarios per OS, one for each version of the software you are targeting, to confirm that your exploit does indeed work on each of the versions of the Agent Tesla panel you state that you are capable of exploiting.